jenkins/format-submission.py

```diff
@@ -0,0 +1,162 @@
+            headers=HEADERS,
+            auth=AUTH,
+        )
+        return response.json()
+    except requests.exceptions.RequestException as e:
+        raise SystemExit(e)
+
+
+def find_mapp_file(attachments):
+    index = 0
+    mapIndex = 0
+    mappNum = 0
+    for attachment in attachments:
+        if attachment.endswith('.mapp'):
+            mappNum += 1
+            mapIndex = index
+        if mappNum > 1:
+            _error_writeback("more than one .mapp file attached")
+        index += 1
+    if mappNum == 0:
+        _error_writeback("no .mapp file attached")
+    return attachments[mapIndex]
+
+
+def format_bundle(mappFile, privacyUrl, documentationUrl, termsOfUseUrl, supportUrl, vendor):
+    try:
+        with zipfile.ZipFile(mappFile, 'r') as zin:
+            # extract into temporary directory
+            tempDir = 'temp'
+            os.system('mkdir {}'.format(tempDir))
+            zin.extractall(tempDir)
+
+        with open(tempDir + '/' + METADATA_FILE, 'r+') as fin:
+            jsoncontents = json.load(fin)
+            jsoncontents['vendor'] = vendor
+            jsoncontents.pop('tags')
+            jsoncontents.update({'metadata': []})
+            jsoncontents['metadata'].append({'tag': 'privacyUrl', 'value': privacyUrl})
+            jsoncontents['metadata'].append({'tag': 'documentationUrl', 'value': documentationUrl})
+            jsoncontents['metadata'].append({'tag': 'termsOfUseUrl', 'value': termsOfUseUrl})
+            jsoncontents['metadata'].append({'tag': 'supportUrl', 'value': supportUrl})
+            
+            # create directory structure: ./http/<vendor>/<exportid> 
+            exportId = jsoncontents['id']
+            httpDir = 'http'
+            vendorDir = '{}/{}'.format(httpDir, vendor)
+            exportDir = '{}/{}/{}'.format(httpDir, vendor, exportId)
+            os.system('mkdir {}'.format(httpDir))
+            os.system('mkdir {}'.format(vendorDir))
+            os.system('mkdir {}'.format(exportDir))        
+            os.system('mv {}/* {}'.format(tempDir, exportDir))
+            os.system('mv {} {}'.format(mappFile, exportDir))    
+
+            fin.seek(0)
+            json.dump(jsoncontents, fin, indent=4)
+            fin.truncate()
+    except: 
+        _error_writeback("missing or malformed .mapp file")
+    
+
+def main():
+    # get the issue in json format and extract the necassary metadata
+    issueJson = get_issue()
+    supportUrl = issueJson['fields'][SUPPORT_URL]
+    documentationUrl = issueJson['fields'][DOCUMENT_URL]
+    privacyUrl = issueJson['fields'][PRIVACY_URL]
+    termsOfUseUrl = issueJson['fields'][TERMS_URL]
+    vendor = issueJson['fields'][VENDOR]
+
+    # download attachemnts from Jira issue
+    attachments = download_attachments()
+    mappFile = find_mapp_file(attachments)
+
+    # format the microapp bundle
+    format_bundle(mappFile, privacyUrl, documentationUrl, termsOfUseUrl, supportUrl, vendor)
+
+
+main()
```

Comment:
Not 100% needed but I think it is nicer to wrap this in `if __name__ == "__main__":`